src/jit/codegenxarch.cpp

```diff
@@ -1813,162 +1813,20 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                     assert(structDesc.eightByteCount == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
 
                     regNumber retReg0 = REG_NA;
-                    emitAttr size0 = EA_UNKNOWN;
-                    unsigned offset0 = structDesc.eightByteOffsets[0];
+                    unsigned __int8 offset0 = 0;
```

Comment:
It is quite inefficient in terms of jit throughput to query VM about struct classification descriptor by passing type handle.  Instead, in lowering we should have detected struct return 2 registers and asked LSRA for two return registers.  Codegen can check whether GT_RETURN has two registers allocated to detect the case of struct return in two registers.